src/thunderbots/software/ai/intent/intent.h

```diff
@@ -2,8 +2,15 @@
 
 #include <string>
 
-// TODO: should this be an enum instead?
-const static std::string MOVE_INTENT_NAME = "Move Intent";
+/**Used for initializing `INTENT_NAME`s and comparisons,
+ * for example getIntent() == XX_INTENT_NAME
+ */
+const static std::string CATCH_INTENT_NAME           = "Catch Intent";
+const static std::string CHIP_INTENT_NAME            = "Chip Intent";
+const static std::string DIRECT_VELOCITY_INTENT_NAME = "Direct Velocity Intent";
+const static std::string KICK_INTENT_NAME            = "Kick Intent";
+const static std::string MOVE_INTENT_NAME            = "Move Intent";
+const static std::string PIVOT_INTENT_NAME           = "Pivot Intent";
```

Comment:
Keeping in-line with how we do this for Primitives, I think this should be at the top of the `.cpp` for each primitive. This is so that anyone who wants the primitive is forced to call `getPrimitiveName()`, rather then using these strings.

Reply:
Just to elaborate on this. You already have the `INTENT_NAME` information in each concrete `Intent` class, so I think you can just remove these since they're redundant.

Reply:
Making that change would break `if (intent->getIntentName() == MOVE_INTENT_NAME)` in rrt.cpp:20. I think that the above line is a clean way to handle the ugliness of identifying intents by a std::string.

Reply:
I'm not exactly sure what you mean about the `std::string`.

The code you have there is fine, but because we already have the Intent names in their respective classes as `static` variables you can do the same thing:
`if(intent->getIntentName() == MoveIntent::INTENT_NAME)`, and this way we aren't storing the names in 2 places.

Reply:
Ok that clears things up. Thanks